### PR TITLE
[docs] add Doxygen group for new TCP Abstractions API

### PR DESCRIPTION
**Pull request by sarah-iot**
Let me know if this name is OK...

### CODE REVIEW

File: doc/ot_api_doc.h [hunk 1]
```diff
@@ -60,6 +60,7 @@
  * @defgroup api-srp                  SRP
  * @defgroup api-ping-sender          Ping Sender
  * @defgroup api-tcp                  TCP
+ * @defgroup api-tcp-ext              TCP Abstractions
```
Comment (Vyrastas) [inline doc/ot_api_doc.h:63]: @jwhui Should we group these together like we do UDP?

Comment (jwhui) [reply doc/ot_api_doc.h:63]: Yes, good idea. I think we should group these together.

Comment (sarah-iot) [reply doc/ot_api_doc.h:67]: @jwhui @Vyrastas No problem, the group is added and tested.

File: doc/ot_api_doc.h [hunk 2]
```diff
@@ -59,8 +59,15 @@
  * @defgroup api-ip6                  IPv6
  * @defgroup api-srp                  SRP
  * @defgroup api-ping-sender          Ping Sender
+ * @defgroup api-tcp-group
```
Comment (Vyrastas) [inline doc/ot_api_doc.h]: I'd add a blank row before this one for readability

**Review by Vyrastas — APPROVED** [review]

File: doc/ot_api_doc.h (hunk 2 above)
Comment (sarah-iot) [reply doc/ot_api_doc.h]: Done.

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**